labellines/core.py

```diff
@@ -122,11 +130,18 @@ def labelLines(
     kwargs : dict, optional
        Optional arguments passed to ax.text
     """
+
     if lines:
         ax = lines[0].axes
     else:
         ax = plt.gca()
 
+    if axis == "y":
+        yaxis = True
+    else:
+        axis = "x"
+        yaxis = False
```

Comment:
Since the check below will be performed at different locations (here and at line 174), I'd suggest factorising it somehow.
I would also recommend explicitly testing the axis to be either `x`, `y` or otherwise fail with an error.

```suggestion
    if axis == "y":
        yaxis = True
    elif axis == "x":
        yaxis = False
    else:
        raise ValueError(r"Got an invalid axis {axis}, expected 'x' or 'y'")
```